lib/functions/modem.js

```diff
@@ -98,31 +98,20 @@ module.exports = function (SerialPort) {
     })
   }
 
-  modem.readSMSById = function (id, callback) {
-    if (callback == undefined) {
-      return new Promise((resolve, reject) => {
-        modem.readSMSById((error, result) => {
-          if (error) {
-            reject(error)
-          } else {
-            resolve(result)
-          }
-        })
-      })
-    }
-    modem.executeCommand(`AT+CMGR=${id}`, function (result) { }, true)
+  modem.readSMSById = function (id) {
+    modem.executeCommand(`AT+CMGR=${id}`, undefined, true)
   }
 
   modem.checkSimMemory = function (callback, priority) {
     if (callback == undefined) {
       return new Promise((resolve, reject) => {
-        modem.checkSimMemory((error, result) => {
+        modem.checkSimMemory((result, error) => {
```

Comment:
as the call back is fired with only one argument in line (130 or 119) the result should be the first argument, this bug caused that before you have been forced to add an empty callback to avoid firing promise `reject` callback.